src/other/tables.jl

```diff
@@ -48,6 +48,23 @@ DataFrame!(x::Vector{<:NamedTuple}) =
                         "`$(typeof(x))` without allocating new columns: use " *
                         "`DataFrame(x)` instead"))
 
+Tables.istable(::Type{<:Union{DataFrameRows,DataFrameColumns}}) = true
+Tables.columnaccess(::Type{<:Union{DataFrameRows,DataFrameColumns}}) = true
+Tables.rowaccess(::Type{<:Union{DataFrameRows,DataFrameColumns}}) = true
+Tables.columns(itr::Union{DataFrameRows,DataFrameColumns}) = Tables.columns(parent(itr))
+Tables.rows(itr::Union{DataFrameRows,DataFrameColumns}) = Tables.rows(parent(itr))
+Tables.schema(itr::Union{DataFrameRows,DataFrameColumns}) = Tables.schema(parent(itr))
+Tables.materializer(itr::DataFrameRows) =
+    eachrow ∘ prefer_singleton_callable(Tables.materializer(parent(itr)))
+Tables.materializer(itr::DataFrameColumns) =
+    eachcol ∘ prefer_singleton_callable(Tables.materializer(parent(itr)))
```

Comment:
If @quinnj comments that we should return here `DataFrameColumns` and not `DataFrame` then we should inherit from `itr` if it was created with `names` positional argument set to `true` or `false`.